pyo3-macros-backend/src/pyclass.rs

```diff
@@ -2203,6 +2218,58 @@ fn pyclass_hash(
     }
 }
 
+fn pyclass_auto_new<'a>(
+    options: &PyClassPyO3Options,
+    cls: &syn::Ident,
+    fields: impl Iterator<Item = &'a &'a syn::Field>,
+    methods_type: PyClassMethodsType,
+    ctx: &Ctx,
+) -> Result<Option<syn::ItemImpl>> {
+    if options.auto_new.is_some() {
+        ensure_spanned!(
+            options.set_all.is_some(), options.hash.span() => "The `auto_new` option requires the `set_all` option.";
+        );
+    }
+    match options.auto_new {
+        Some(opt) => {
+            if matches!(methods_type, PyClassMethodsType::Specialization) {
+                bail_spanned!(opt.span() => "`auto_new` requires the `multiple-pymethods` feature.");
+            }
```

Comment:
We don't need this restriction (at the cost of a slightly more complicated implementation) - see `generate_protocol_slot` and how it's used for `__str__`.